src/main/java/net/mcjukebox/plugin/bukkit/managers/RegionManager.java

```diff
@@ -32,6 +36,23 @@ public void addRegion(String ID, String URL){
     }
 
     public void removeRegion(String ID){
+        ShowManager showManager = MCJukebox.getInstance().getShowManager();
+        HashMap<UUID, String> playersInRegion = MCJukebox.getInstance().getRegionListener().getPlayerInRegion();
+
+        for (HashMap.Entry<UUID, String> entry: playersInRegion.entrySet()) {
+            UUID uuid = entry.getKey();
+            String regionID = entry.getValue();
+
+            if (regionID.equals(ID)) {
+
+                if (regions.get(ID).charAt(0) == '@') {
+                    showManager.getShow(regions.get(ID)).removeMember(Bukkit.getPlayer(uuid));
+                } else {
+                    JukeboxAPI.stopMusic(Bukkit.getPlayer(uuid));
+                }
```

Comment:
After this line, we know the music playing to them has been stopped so we should remove them from the playersInRegion hashmap.